Upstream-Status: Backport [https://github.com/eProsima/Fast-DDS/commit/e1524ac973b759d4756bbe13df80a34344eb3d29]

Signed-off-by: Rob Woolley <rob.woolley@windriver.com>

From e1524ac973b759d4756bbe13df80a34344eb3d29 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mario=20Dom=C3=ADnguez=20L=C3=B3pez?=
 <116071334+Mario-DL@users.noreply.github.com>
Date: Fri, 9 May 2025 11:55:39 +0200
Subject: [PATCH] Address some compilation warnings with GCC latest (#5796)

* Refs #23110: Add cstdint header for uint8_t

Signed-off-by: Mario Dominguez <mariodominguez@eprosima.com>

* Refs #23110: Remove unused using

Signed-off-by: Mario Dominguez <mariodominguez@eprosima.com>

* Refs #23110: Add missing cstdint to more headers

Signed-off-by: cferreiragonz <carlosferreira@eprosima.com>

---------

Signed-off-by: Mario Dominguez <mariodominguez@eprosima.com>
Signed-off-by: cferreiragonz <carlosferreira@eprosima.com>
Co-authored-by: cferreiragonz <carlosferreira@eprosima.com>
---
 .../fastdds/topic/DDSSQLFilter/DDSFilterCompoundCondition.hpp    | 1 +
 src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterPredicate.hpp        | 1 +
 src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterValue.hpp            | 1 +
 4 files changed, 3 insertions(+), 1 deletion(-)

Index: fastrtps-2.6.9-1/src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterCompoundCondition.hpp
===================================================================
--- fastrtps-2.6.9-1.orig/src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterCompoundCondition.hpp
+++ fastrtps-2.6.9-1/src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterCompoundCondition.hpp
@@ -19,6 +19,7 @@
 #ifndef _FASTDDS_TOPIC_DDSSQLFILTER_DDSFILTERCOMPOUNDCONDITION_HPP_
 #define _FASTDDS_TOPIC_DDSSQLFILTER_DDSFILTERCOMPOUNDCONDITION_HPP_
 
+#include <cstdint>
 #include <memory>
 
 #include "DDSFilterCondition.hpp"
Index: fastrtps-2.6.9-1/src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterPredicate.hpp
===================================================================
--- fastrtps-2.6.9-1.orig/src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterPredicate.hpp
+++ fastrtps-2.6.9-1/src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterPredicate.hpp
@@ -19,6 +19,7 @@
 #ifndef _FASTDDS_TOPIC_DDSSQLFILTER_DDSFILTERPREDICATE_HPP_
 #define _FASTDDS_TOPIC_DDSSQLFILTER_DDSFILTERPREDICATE_HPP_
 
+#include <cstdint>
 #include <memory>
 
 #include "DDSFilterCondition.hpp"
Index: fastrtps-2.6.9-1/src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterValue.hpp
===================================================================
--- fastrtps-2.6.9-1.orig/src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterValue.hpp
+++ fastrtps-2.6.9-1/src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterValue.hpp
@@ -19,6 +19,7 @@
 #ifndef _FASTDDS_TOPIC_DDSSQLFILTER_DDSFILTERVALUE_HPP_
 #define _FASTDDS_TOPIC_DDSSQLFILTER_DDSFILTERVALUE_HPP_
 
+#include <cstdint>
 #include <memory>
 #include <regex>
 
